cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -0,0 +1,102 @@
+"Contentstore Views"
+
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import status
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from common.lib.xmodule.xmodule.course_module import get_available_providers
+from contentstore.views.course import get_course_and_check_access
+from models.settings.course_metadata import CourseMetadata
+from openedx.core.lib.api.view_utils import view_auth_classes
+from xmodule.modulestore.django import modulestore
+
+from contentstore.rest_api.v1.serializers import ProctoredExamSettingsMainSerializer
+
+
+@api_view(['GET'])
+@view_auth_classes()
+def proctored_exam_settings(request, course_id):
```

Comment:
Why implement this as a function instead of an `APIView` class? I thought you need this view to also include POST handler later, correct?

Reply:
I didn't think it was necessary to make this a class based view until there was a demonstrated need for it. We can handle a POST request here by checking the value of `request.method`.

But I can make this a class based view in anticipation of the POST handler.

Reply:
Since I'm working off this to add POST already, I can take care of refactoring this into a class if you haven't already started.

Reply:
That would be great. Thanks, Zach.